frontend/packages/dev-console/src/components/formik-fields/multi-column-field/MultiColumnField.tsx

```diff
@@ -33,8 +33,7 @@ const MultiColumnField: React.FC<MultiColumnFieldProps> = ({
           {fieldValue.length > 0 &&
             fieldValue.map((value, index) => (
               <MultiColumnFieldRow
-                // eslint-disable-next-line react/no-array-index-key
-                key={`multi-column-field-row-${index}`} // There is no other usable value for key prop in this case.
+                key={`multi-column-field-row-${JSON.stringify(value)}`}
```

Comment:
I think there may be an issue with this - after I wrote the answer below, I started trying to think of what types `value` could be.... it occurs to me, it's probably an object for the entire row, no? If that's the case, we're likely looking at massive remounts. Every time some value is changed it remounts the whole row.

**Original Thought:**
I think this is mostly okay... This is a generic component, so technically, you could have multiple same-value items. If this conflict hits, it won't render the 2nd one and will log the error in the console; "children must have unique react keys" or something to that extent.

Reply:
Yeah, I ran into the remount issue while testing with `InputField` which affects performance drastically. That is why I had to cancel lgtm. Didn't notice the performance issue earlier as I was testing with dropdowns primarily. 

Trying to find a better way to make the keys unique. But don't think we have much option in this case. Can't even use uuids as it would affect performance similar to what using values does.

Reply:
Is it possible to pass in some sort of `keyId` prop to the component? That way the mounting code can specific which part of the object is the key. If one is not provided, we could use index as a fallback?

Reply:
We need the keys to be unique for each row. Even if we send a `keyId` prop from parent component, it will be unique for the whole `MultiColumnField` component and not for each rows.

Reply:
Sorry - I think we are having a miscommunication, here is a fake snippet to explain what I mean:

```
{fieldValue.length > 0 &&
  fieldValue.map((value, index) => (
    <MultiColumnFieldRow
      key={`multi-column-field-row-${values[props.keyId]}`} // << It's a prop identifier
      name={name}
      rowIndex={index}
      onDelete={() => remove(index)}
      isReadOnly={isReadOnly}
    >
      {children}
    </MultiColumnFieldRow>
  ))}
```

Reply:
So, `fieldValue` is an array of objects and value would be something like `{ name: foo, type: bar }`. You mean to say we add `keyId` in `value` object? The problem in that case would be we need to again generate those keyIds for each new row that user adds as well.

Reply:
And if you meant using the value for a key like `name` in an object `{name: foo, type: bar}`, we'll have the same remount issue as the value of `name` will change on every keystroke of user.

Reply:
So, react keys are almost always an 'id' field or something similar. They are used to keep track of lists of items. It should ideally come from data.

What I am suggesting here is that we allow the `MultiColumnField` component to take a prop-name (my suggestion was `keyId`) to allow for the caller of the component (whom theoretically would know their data) to make unique ids.

Now, the caller can select a prop from existing data, or they can create one by combining props together into a new prop (to which they specify as their `keyId`).

Surely the data we are getting in View/Edit Permissions form is not _just_ name and permission level. If it is, perhaps on retrieve of fetch we generate one (which can be an index number for all matching sakes) which does not change for the life of the page. Each time a new one is added, we generate a new one.

The key content does not matter, it just needs to be unique.